common/src/main/java/dev/isxander/yacl3/gui/controllers/ColorPickerElement.java

```diff
@@ -0,0 +1,435 @@
+            mouseDown = true;
+            hueSliderDown = false;
+            satLightGradientDown = false;
+            alphaSliderDown = false;
+            setColorFromMouseClick(mouseX, mouseY);
+            return true;
+        } else if(entryWidget.isMouseOver(mouseX, mouseY)) {
+            return entryWidget.mouseClicked(mouseX, mouseY, button);
+        } else {
+            if(Minecraft.getInstance().screen instanceof PopupControllerScreen popupControllerScreen) {
+                popupControllerScreen.onClose();
+            }
+            return false;
+        }
+    }
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        mouseDown = false;
+        return false;
+    }
+    @Override
+    public boolean isMouseOver(double mouseX, double mouseY) {
+        //Checks if the mouse is either over the color picker or the color controller
+        //The addition/subtraction of the outline and extra 3 pixels is to account for both the outline and the background
+        if (mouseX >= colorPickerDim.x() - outline - 3 && mouseX <= colorPickerDim.xLimit() + outline + 3
+                && mouseY >= colorPickerDim.y() - outline - 3 && mouseY <= colorPickerDim.yLimit() + outline + 3) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean mouseDragged(double mouseX, double mouseY, int button, double deltaX, double deltaY) {
+        if(mouseDown || isMouseOver(mouseX, mouseY)) {
+            setColorFromMouseClick(mouseX, mouseY);
+            return true;
+        }
+        return entryWidget.mouseDragged(mouseX, mouseY, button, deltaX, deltaY);
+    }
+
+    @Override
+    public boolean charTyped(char chr, int modifiers) {
+        //Done to allow for typing whilst the color picker is visible
+        charTyped = true;
+        return entryWidget.charTyped(chr, modifiers);
+    }
+
+    @Override
+    public void setDimension(Dimension<Integer> dim) {
+        super.setDimension(dim);
+
+        int colorPickerHeight = (dim.height() * 2) + 7;
+        int colorPickerX = dim.centerX() - getXPadding() * 2;
+        int colorPickerY = dim.y() - colorPickerHeight - sliderHeight;
+        int alphaSliderHeight = 0;
+        if(controller.allowAlpha()) {
+            alphaSliderHeight = sliderHeight + outline + paddingY;
+            colorPickerHeight += alphaSliderHeight;
+            colorPickerY -= alphaSliderHeight;
+        }
+
+        //A single dimension for the entire color picker as a whole
+        //Division is used for the main color preview, saturation/light picker, and hue slider to determine their dimensions
+        colorPickerDim = Dimension.ofInt(colorPickerX, colorPickerY, dim.xLimit() - colorPickerX, colorPickerHeight);
+
+        //Add anything after the colorPickerDim.y() to reduce the "strictness" of the detection
+        if(colorPickerDim.y() < screen.tabArea.top()){
+            colorPickerY = dim.yLimit() + sliderHeight;
```

Comment:
Small nitpick: there's not really a reason to change the `colorPickerY` variable at this point, as it's only used in the very next line. The order of operations here confused me a little when trying to read the code. Personally, I would move the construction of `colorPickerDim` below this condition (getting rid of the `setY` call) to mirror the `allowAlpha` branch.

Reply:
Fixed, thanks.